sentry-apache-http-client-5/src/main/java/io/sentry/transport/apache/ApacheHttpClientTransport.java

```diff
@@ -0,0 +1,147 @@
+  private final @NotNull SentryOptions options;
+  private final @NotNull RequestDetails requestDetails;
+  private final @NotNull CloseableHttpAsyncClient httpclient;
+  private final @NotNull RateLimiter rateLimiter;
+  private final @NotNull AtomicInteger currentlyRunning;
+
+  public ApacheHttpClientTransport(
+      final @NotNull SentryOptions options,
+      final @NotNull RequestDetails requestDetails,
+      final @NotNull CloseableHttpAsyncClient httpclient,
+      final @NotNull RateLimiter rateLimiter) {
+    this(options, requestDetails, httpclient, rateLimiter, new AtomicInteger());
+  }
+
+  ApacheHttpClientTransport(
+      final @NotNull SentryOptions options,
+      final @NotNull RequestDetails requestDetails,
+      final @NotNull CloseableHttpAsyncClient httpclient,
+      final @NotNull RateLimiter rateLimiter,
+      final @NotNull AtomicInteger currentlyRunning) {
+    this.options = Objects.requireNonNull(options, "options is required");
+    this.requestDetails = Objects.requireNonNull(requestDetails, "requestDetails is required");
+    this.httpclient = Objects.requireNonNull(httpclient, "httpclient is required");
+    this.rateLimiter = Objects.requireNonNull(rateLimiter, "rateLimiter is required");
+    this.currentlyRunning =
+        Objects.requireNonNull(currentlyRunning, "currentlyRunning is required");
+    this.httpclient.start();
+  }
+
+  @Override
+  @SuppressWarnings("FutureReturnValueIgnored")
+  public void send(SentryEnvelope envelope, Object hint) throws IOException {
+    if (isSchedulingAllowed()) {
+      final SentryEnvelope filteredEnvelope = rateLimiter.filter(envelope, hint);
+
+      if (filteredEnvelope != null) {
+        currentlyRunning.incrementAndGet();
+
+        try (final ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+            final GZIPOutputStream gzip = new GZIPOutputStream(outputStream)) {
+          options.getSerializer().serialize(filteredEnvelope, gzip);
+
+          final SimpleHttpRequest request =
+              SimpleHttpRequests.post(requestDetails.getUrl().toString());
+          request.setBody(outputStream.toByteArray(), ContentType.APPLICATION_JSON);
+          request.setHeader("Content-Encoding", "gzip");
+          request.setHeader("Content-Type", "application/x-sentry-envelope");
+          request.setHeader("Accept", "application/json");
+
+          for (Map.Entry<String, String> header : requestDetails.getHeaders().entrySet()) {
+            request.setHeader(header.getKey(), header.getValue());
+          }
+
+          if (options.getLogger().isEnabled(DEBUG)) {
+            options.getLogger().log(DEBUG, "Currently running %d requests", currentlyRunning.get());
+          }
+
+          httpclient.execute(
+              request,
+              new FutureCallback<SimpleHttpResponse>() {
+                @Override
+                public void completed(SimpleHttpResponse response) {
+                  currentlyRunning.decrementAndGet();
```

Comment:
should this be the last line to be called? theoretically there's still something happening.

Reply:
There is but it means that the connection is not occupied anymore. I don't see any risk here.